hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -328,6 +346,14 @@ private void setHmsTableParameters(
       Set<String> obsoleteProps,
       boolean hiveEngineEnabled,
       Map<String, String> summary) {
+
+    setHmsParameters(
```

Comment:
Wouldn't the different ordering of this and the metadata translation cause changes?

Reply:
I am not able to relate to any impacts of re-ordering. Could you please help me with some examples ?

Reply:
I think both moving before, and after is a change in behaviour.

With the current code, the user could remove any props, expect the `TABLE_TYPE_PROP`, `METADATA_LOCATION_PROP`, `PREVIOUS_METADATA_LOCATION_PROP` and schema. (Not sure why UUID is removable - CC: @flyrain)

If we put the obsolate before, then none of them is removable, if we put obsolate after, then all of them is removable.